Sources/web3swift/Web3/Web3+EIP1559.swift

```diff
@@ -0,0 +1,177 @@
+
+        // Check that gas limit below upper bound
+        if BigUInt(diff) >= limit { return false }
+
+        // Check that gas limit above lower bound
+        if currentGasLimit < Web3.MinGasLimit { return false }
+
+        return true
+    }
+
+    /// Method to check is given block valid EIP-1559 block
+    ///
+    /// Verifies some header attributes which were changed in EIP-1559
+    ///
+    /// - gas limit check
+    /// - basefee check
+    ///
+    /// - Parameters:
+    ///   - parent: Previous Block
+    ///   - current: Current block
+    /// - Returns: True or false if block is EIP-1559 or not
+    func isEip1559Block(parent: Block, current: Block) -> Bool {
+        let parentGasLimit = parent.chainVersion >= .London ? parent.gasLimit : parent.gasLimit * Web3.ElasticityMultiplier
+
+        guard verifyGasLimit(parentGasLimit: parentGasLimit, currentGasLimit: current.gasLimit) else { return false }
+
+        guard calcBaseFee(parent) == current.baseFeePerGas else { return false }
+
+        return true
+    }
+
+    /// Calculates base fee amount
+    ///
+    /// You should pass **parent** block to that method to calculate **expected** baseFee for currently
+    /// processing by blockchain `Block`
+    ///
+    /// Calculation for current `Block` based on parents block object only
+    ///
+    /// If passed block isn't `ChainVersion.London` one will return
+    ///
+    /// - Parameter parent: Parent `Block`
+    /// - Returns: Amount of expected base fee for current `Block`
+    func calcBaseFee(_ parent: Block) -> BigUInt {
+        // If given blocks ChainVersion is lower than London — always returns InitialBaseFee
+        guard parent.chainVersion >= .London else { return Web3.InitialBaseFee }
+
+        let parentGasTarget = parent.gasLimit / Web3.ElasticityMultiplier
+
+        if parent.gasUsed > parentGasTarget {
+            // If the parent block used more gas than its target, the baseFee should increase.
+            let gasUsedDelta = parent.gasUsed - parentGasTarget
+            let baseFeePerGasDelta = max(parent.baseFeePerGas * gasUsedDelta / parentGasTarget / Web3.BaseFeeChangeDenominator, 1)
+            let expectedBaseFeePerGas = parent.baseFeePerGas + baseFeePerGasDelta
+
+            return expectedBaseFeePerGas
+        } else if parent.gasUsed < parentGasTarget  {
+            // Otherwise if the parent block used less gas than its target, the baseFee should decrease.
+            let gasUsedDelta = parentGasTarget - parent.gasUsed
+            let baseFeePerGasDelta = parent.baseFeePerGas * gasUsedDelta / parentGasTarget / Web3.BaseFeeChangeDenominator
+            let expectedBaseFeePerGas = parent.baseFeePerGas - baseFeePerGasDelta
+
+            return expectedBaseFeePerGas
+        } else {
+            // If the parent gasUsed is the same as the target, the baseFee remains unchanged.
+            return parent.baseFeePerGas
+        }
+    }
+}
+
+public extension Web3 {
```

Comment:
Would these ChainVersion/Block numbers not be different for every network? My worry is that this code will only work on Etereum/mainnet. Not sure, but possibly even the Ethereum test nets have different fork block numbers.